datadog/fwprovider/data_source_datadog_app.go

```diff
@@ -0,0 +1,78 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var _ datasource.DataSource = &appDataSource{}
+
+type appDataSource struct {
+	Api  *datadogV2.AppBuilderApi
+	Auth context.Context
+}
+
+func NewDatadogAppDataSource() datasource.DataSource {
+	return &appDataSource{}
+}
+
+func (d *appDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData := request.ProviderData.(*FrameworkProvider)
+	d.Api = providerData.DatadogApiInstances.GetAppBuilderApiV2()
+	d.Auth = providerData.Auth
+}
+
+func (d *appDataSource) Metadata(_ context.Context, request datasource.MetadataRequest, response *datasource.MetadataResponse) {
+	response.TypeName = "app"
```

Comment:
maybe worth naming the resource `app_builder_app` across this PR since `app` is such an overloaded word